core/src/main/java/org/mule/runtime/core/el/DataWeaveExpressionLanguageAdaptor.java

```diff
@@ -98,8 +103,17 @@ public void addGlobalBindings(BindingContext bindingContext) {
 
   @Override
   public TypedValue evaluate(String expression, Event event, BindingContext context) {
-    BindingContext.Builder contextBuilder = bindingContextBuilderFor(event, context);
-    return sanitizeAndEvaluate(expression, exp -> expressionExecutor.evaluate(exp, contextBuilder.build()));
+    String sanitized = sanitize(expression);
+    if (isPayloadExpression(sanitized)) {
+      return event.getMessage().getPayload();
+    } else {
+      BindingContext.Builder contextBuilder = bindingContextBuilderFor(event, context);
+      return sanitizeAndEvaluate(sanitized, exp -> expressionExecutor.evaluate(exp, contextBuilder.build()));
```

Comment:
We should skip sanitising it again. Since exception handling is also done the I'm thinking we need to split that method to reuse the exception stuff here.